nixos/modules/virtualisation/qemu-vm.nix

```diff
@@ -282,16 +308,50 @@ in
           '';
       };
 
+    virtualisation.shareExchangeDir =
+      mkOption {
+        default = true;
+        type = types.bool;
+        description =
+          ''
+            Share /tmp/xchg with the host.
+          '';
+      };
+
+    virtualisation.shareSharedDir =
+      mkOption {
+        default = true;
+        type = types.bool;
+        description =
+          ''
+            Share /tmp/shared with the host.
+          '';
+      };
```

Comment:
What's the difference of these?

Reply:
I don't know.

By default they point at the same directory in the host.  The test framework points them at different directories by setting `SHARED_DIR`.

History:
* Aug 2011 d75efe4aa184c4fa89241cf3df2f04616f3ba93f edolstra: Introduced /tmp/xchg: "For security, don't mount the entire host filesystem."
* Aug 2012 a025e7e7e287c2539d883bd40a769ae29f9087ff edolstra: Introduced /tmp/shared: "Provide a common share between VMs to allow easy communication"
* May 2020 d85dc4f6902819be8277d1d65414e51bfbef0ac6 erikarvstedt: Mentions "xchg is advertised as a bidirectional exchange dir."

Current uses I could find:
| Mount | Purpose | Location |
| --- | --- | --- |
| xchg | coverage-data | nixos/modules/testing/test-instrumentation.nix nixos/lib/test-driver/test-driver.pl |
| xchg | runInMachine | nixos/lib/testing.nix |
| shared | copy_from_host copy_from_vm | nixos/lib/test-driver/test-driver.py |


I just want a lever to turn them off.  I guess it makes more sense to be one lever rather than two.

I combined these two options into one and added a note about what these directories are currently used for in the option description.

Reply:
@erikarvstedt @edolstra can you provide some insight here?

Reply:
We can probably get rid of `coverage-data`. AFAIK we're not doing coverage runs in VMs anymore (and IIRC it was only used in some toy examples like the Quake 3 test anyway).

Reply:
We can probably also eliminate `copy_from_*` since it's only used in 3 places, which can be replaced by `machine.execute("cat ...")`.

Reply:
> We can probably get rid of `coverage-data`. AFAIK we're not doing coverage runs in VMs anymore (and IIRC it was only used in some toy examples like the Quake 3 test anyway).

I updated the list in https://github.com/NixOS/nixpkgs/issues/72828, and filed https://github.com/NixOS/nixpkgs/pull/97231 removing the remaining `coverage-data` references - PTAL.